core/src/main/java/org/jdbi/v3/core/annotation/JdbiProperty.java

```diff
@@ -0,0 +1,37 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.jdbi.v3.core.annotation;
+
+import java.lang.annotation.ElementType;
+import java.lang.annotation.Retention;
+import java.lang.annotation.RetentionPolicy;
+import java.lang.annotation.Target;
+
+import org.jdbi.v3.core.internal.AnnotationFactory;
+import org.jdbi.v3.meta.Beta;
+
+/**
+ * Configure aspects of discovered bean / pojo properties.
+ * Currently only works with reflective bean and pojo mapping.
+ */
+@Beta
+@Target(ElementType.METHOD)
+@Retention(RetentionPolicy.RUNTIME)
+public @interface JdbiProperty {
```

Comment:
What would you think of an `@Unmappable` annotation instead?

Reply:
So, I was doing this for two reasons:

It gives us a natural place to add more properties later like `@JdbiProperty(name="custom_name")`

Allow you to re-enable mapping in subclasses.  This is a practice Jackson has: https://fasterxml.github.io/jackson-annotations/javadoc/2.0.0/com/fasterxml/jackson/annotation/JsonValue.html#value()

Reply:
What about `@Mappable(false)` then?

Reply:
Do you not like the ability to extend the same annotation later with more properties?  `Mappable` means it will only ever control mapping behavior, I was imagining it more like `@JsonProperty` which has a number of knobs.

Reply:
Maybe I misunderstand your intent, but I don't see the need for a `@JdbiProperty` with a name attribute, since we already have `@ColumnName`.

Reply:
OK, we'll give it a shot.